### PR TITLE
Fix/pn 3802

### PR DESCRIPTION
**Pull request by leleOFA**
## Short description
fix see ToS and Privacy policy with an user unauthorized

## List of changes proposed in this pull request
- create proposed hook
- used in  fees page 

## How to test
when click in menù navigation you don't see login page

### CODE REVIEW

File: packages/pn-pa-webapp/src/pages/TermsOfService.page.tsx [hunk 1]
```diff
@@ -12,19 +14,27 @@ declare const OneTrust: {
...
 
 const TermsOfServicePage = () => {
+  const [contentLoaded, setContentLoaded] = useState(false);
+
   useEffect(() => {
     if (ONE_TRUST_PORTAL_CDN_TOS) {
       OneTrust.NoticeApi.Initialized.then(function () {
         OneTrust.NoticeApi.LoadNotices([ONE_TRUST_PORTAL_CDN_TOS], false);
       });
+      setContentLoaded(true);
```
Comment (AndreaCimini90) [inline packages/pn-pa-webapp/src/pages/TermsOfService.page.tsx:24]: Same as before

File: packages/pn-personafisica-webapp/src/pages/TermsOfService.page.tsx [hunk 2]
```diff
@@ -12,19 +14,27 @@ declare const OneTrust: {
...
 
 const TermsOfServicePage = () => {
+  const [contentLoaded, setContentLoaded] = useState(false);
+
   useEffect(() => {
     if (ONE_TRUST_PORTAL_CDN_TOS) {
       OneTrust.NoticeApi.Initialized.then(function () {
         OneTrust.NoticeApi.LoadNotices([ONE_TRUST_PORTAL_CDN_TOS], false);
       });
+      setContentLoaded(true);
```
Comment (AndreaCimini90) [inline packages/pn-personafisica-webapp/src/pages/TermsOfService.page.tsx:24]: Same as before

**Review by AndreaCimini90 — CHANGES_REQUESTED** [review]

File: packages/pn-commons/src/hooks/useDocumentationUnauthorized.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { useLayoutEffect } from 'react';
+
+export const useDocumentationUnauthorized = (contentLoaded: boolean, route: any) => {
+  useLayoutEffect(() => {
+    setTimeout(() => {
+      const links = document.querySelectorAll('.otnotice-content a');
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-commons/src/hooks/useDocumentationUnauthorized.ts:6]: do it

File: packages/pn-pa-webapp/src/pages/PrivacyPolicy.page.tsx [hunk 4]
```diff
@@ -11,14 +12,19 @@ declare const OneTrust: {
...
 
 const PrivacyPolicyPage = () => {
+  const [contentLoaded, setContentLoaded] = useState(false);
+
   useEffect(() => {
     if (ONE_TRUST_PORTAL_CDN_PP) {
       OneTrust.NoticeApi.Initialized.then(function () {
         OneTrust.NoticeApi.LoadNotices([ONE_TRUST_PORTAL_CDN_PP], false);
       });
+      setContentLoaded(true);
```
[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-pa-webapp/src/pages/PrivacyPolicy.page.tsx:22]: do it for all

File: packages/pn-commons/src/hooks/useDocumentationUnauthorized.ts [hunk 5]
```diff
@@ -0,0 +1,16 @@
+import { useLayoutEffect } from 'react';
+
+export const useDocumentationUnauthorized = (contentLoaded: boolean, route: any) => {
```
[the remark this answers is not in the document]
Comment (ZorbaDimatteo) [reply packages/pn-commons/src/hooks/useDocumentationUnauthorized.ts]: @AndreaCimini90 I like it!

[the remark this answers is not in the document]
Comment (leleOFA) [reply packages/pn-commons/src/hooks/useDocumentationUnauthorized.ts]: i rename it

**Review by AndreaCimini90 — APPROVED** [review]

**Review by AndreaCimini90 — APPROVED** [review]

**Pull request merged**